sdks/python/apache_beam/transforms/core.py

```diff
@@ -2354,14 +2356,28 @@ def __init__(
     self._partial = partial
     self._on_failure_callback = on_failure_callback
 
+    # Wrap process and expose any top level state params so that process can
+    # handle state.
+    @wraps(self._fn.process)
+    def process_wrapper(*args, **kwargs):
+      return self.custom_process(*args, **kwargs)
+    self.process = process_wrapper
+    for i in dir(self._fn):
+      member = getattr(self._fn, i)
+      if isinstance(member, StateSpec):
+        setattr(self, i, member)
+
+    from apache_beam.runners.common import DoFnSignature
+    raise ValueError('is stateful: ' + str(DoFnSignature(self).is_stateful_dofn()) + str(inspect.signature(self.process)))
```

Comment:
This is returning that it is not stateful, but it is returning the expected process signature. Something is a little off here.